internal/utils/validate.go

```diff
@@ -113,61 +105,3 @@ func ParseReference(mergedIndexes *index.MergedIndexes, name string) (string, er
 
 	return ref, nil
 }
-
-// CheckRegistryConnection checks whether the registry implement Docker Registry API V2 or
-// OCI Distribution Specification. It also checks authentication if credentials are not empty.
-func CheckRegistryConnection(ctx context.Context, cred *auth.Credential, regName string, printer *output.Printer) error {
-	sp, _ := printer.Spinner.Start(fmt.Sprintf("INFO: Checking connection to remote registry %q", regName))
-
-	if reflect.DeepEqual(*cred, auth.EmptyCredential) {
-		if err := checkRegistryUnauthenticated(ctx, regName); err != nil {
-			return err
-		}
-		sp.Success(fmt.Sprintf("Remote registry %q implements docker registry API V2", regName))
-		printer.Verbosef("Continuing without authentication, no user credentials provided")
-		return nil
-	}
-
-	client := authn.NewClient(authn.WithCredentials(cred))
-
-	// Ensure credentials are valid.
-	registry, err := remote.NewRegistry(regName)
-	if err != nil {
-		return err
-	}
-
-	registry.Client = client
-	if err = registry.Ping(ctx); err != nil {
-		return err
-	}
-
-	sp.Success(fmt.Sprintf("Remote registry %q implements docker registry API V2", regName))
-	printer.Verbosef("Proceeding as user %q", cred.Username)
-
-	return nil
-}
-
-func checkRegistryUnauthenticated(ctx context.Context, regName string) error {
```

Comment:
From my experiments, this function was useless, since if we have EmptyCredentials, no bearer is appended as authorization header, which is totally equivalent to what CheckRegistryConnection will do with empty credentials.

Reply:
I think we still need it. If the user is not authenticated commands interacting with a remote registry will fail with error: ` unexpected status code 403: denied: requested access to the resource is denied`

Reply:
I am thinking about this again. Since this function is always called in the flow for push and pull, for me it doesn't make sense checking the connection. The HTTP error we will receive from the first request for push and pull will exactly be the same. It makes much more sense, instead, when checking credentials for login: if we can connect to registry through the supplied credentials, then we save them. This is also the same thing that oras-cli does, the ping is only performed in the login, not at every push and pull operation. That's why I am deleting the check for ClientForRegistry function.

Reply:
The check is back, ported into the registry package and also improved by verifying correct header is set. This was needed to perform the additional check on the implementation of v2 (the above comment addressed only authentication). Thank you!